GermanWordle/R/Week2functions.R

```diff
@@ -0,0 +1,41 @@
+# Week 2 functions
+
+#' @export
+source("Wordlist.R")
+
+#' @export
+target <- sample(german_words, 1)
+
+#' @title Compare Words
+#' @description This function compares whether the letters in the target word are the same as in the guessed word
+#' @param target_str character. Should be a word with the same length as the guessed word
+#' @param guess_str character. Should be a word with the same length as the target word
+#' 
+#' @return Shows whether the letter is in the word and on the correct place, in the word but incorrect place, or not in the word
+#' @examples
+#' # Compare two words
+#' compare_words("haus","maus")
+#' @export
+compare_words <- function(target_str, guess_str) {
+  if (nchar(target_str) != nchar(guess_str)) {
+    stop("target and guess string must be the same length.")
+  }
+  
+  target <- strsplit(target_str, "")[[1]]
+  guess <- strsplit(guess_str, "")[[1]]
+  result <- character(nchar(guess_str))
+  
```

Comment:
Just a suggestion but you can add comments to your function and elaborate a bit more, and also check whether the word the users guessed is in the wordlist.